fed/mosaic-sumo/src/main/java/org/eclipse/mosaic/fed/sumo/ambassador/SumoAmbassador.java

```diff
@@ -207,9 +207,8 @@ private void receiveInteraction(VehicleRegistration interaction) {
      */
     private void receiveInteraction(VehicleRouteRegistration interaction) throws InternalFederateException {
         VehicleRoute newRoute = interaction.getRoute();
-        routeCache.put(newRoute.getId(), newRoute);
-        if (!sumoKnownRoutes.contains(newRoute.getId())) {
-            sumoKnownRoutes.add(newRoute.getId());
+        if (!routeCache.containsKey(newRoute.getId())) {
+            routeCache.put(newRoute.getId(), newRoute);
             traci.getRouteControl().addRoute(newRoute.getId(), newRoute.getEdgeIdList());
             log.debug("received newly propagated route {}", newRoute.getId());
```

Comment:
Could you add the edgeIdList() of the route here in the debug-statement? (as we cannot look into temporary generated rou.xml anymore for debugging purposes)

Reply:
Done.